tool/tsh/tsh.go

```diff
@@ -1511,7 +1568,33 @@ func sortedLabels(labels map[string]string) string {
 	return strings.Join(append(result, namespaced...), ",")
 }
 
-func showApps(apps []types.Application, active []tlsca.RouteToApp, verbose bool) {
+func showApps(apps []types.Application, active []tlsca.RouteToApp, format string, verbose bool) error {
+	format = strings.ToLower(format)
+	switch format {
+	case teleport.Text:
+		showAppsAsText(apps, active, verbose)
+	case teleport.JSON, teleport.YAML:
+		if apps == nil {
+			apps = []types.Application{}
+		}
```

Comment:
Will Marshal panic is apps are nil?

Reply:
it won't panic, but it will serialize `nil` as `null` instead of `[]`.